crates/sqlexec/src/planner/session_planner.rs

```diff
@@ -458,6 +459,17 @@ impl<'a> SessionPlanner<'a> {
                         source: Box::new(e),
                     })?;
 
+                let url = url::Url::parse(&url).unwrap();
+
+                let store = access.store().unwrap();
+                let base_location = access.base_location();
+
+                println!("registering object store for {}", url);
+                self.ctx
+                    .get_df_state()
+                    .runtime_env()
+                    .register_object_store(&url, store);
```

Comment:
ideally, the object store registry should fetch this from the catalog, instead of having to manually insert it like this.